lib/instrumentation/aws-sdk/v3/dynamodb.js

```diff
@@ -72,21 +72,66 @@ function dynamoMiddleware(shim, config, next, context) {
   }
 }
 
-const dynamoMiddlewareConfig = {
-  middleware: dynamoMiddleware,
-  init(shim) {
-    shim.setDatastore(shim.DYNAMODB)
-    return true
-  },
-  type: InstrumentationDescriptor.TYPE_DATASTORE,
-  config: {
-    name: 'NewRelicDynamoMiddleware',
-    step: 'initialize',
-    priority: 'high',
-    override: true
+/**
+ * Wraps the deserialize middleware step to add the
+ * cloud.resource_id segment attributes for the AWS command
+ *
+ * @param {Shim} shim
+ * @param {Object} config AWS command configuration
+ * @param {function} next next function in middleware chain
+ * @returns {function}
+ */
+function resourceIdMiddlerware(shim, config, next) {
+  return async function wrappedResourceIdMiddlerware(args) {
+    let region
+    try {
+      region = await config.region()
+      const segment = shim.getSegment()
+
+      const accountId = shim.agent.config.cloud.aws.account_id
```

Comment:
I _think_ we should only be setting this `cloud.resource_id` if accountId is truthy, otherwise this attribute will be useless for making the entity relationship

Reply:
Got it. I've updated it to check for accountId before setting the resource id